shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -302,6 +305,13 @@ - (BOOL)runWithEntrypoint:(NSString*)entrypoint {
     flutterArguments.aot_data = _aotData;
   }
 
+  // Only create a Compositor if we have a ViewController.
+  if (_viewController) {
+    FlutterCompositor compositor = {};
```

Comment:
Lets move compositor to be a FlutterEngine initialzer param `_compositor`, that will make the lifecycle clear.

Reply:
Would it be better to just leave a comment stating that the compositor gets deep copied in embedder.cc?

Reply:
Went with the comment for now, let me know if you don't think this is sufficient